src/soundio/soundmanager.cpp

```diff
@@ -363,7 +388,77 @@ SoundDeviceStatus SoundManager::setupDevices() {
 
     // pair is isInput, isOutput
     QVector<DeviceMode> toOpen;
-    bool haveOutput = false;
+
+    // Get all outputs for each device
+    QHash<SoundDevicePointer, QList<AudioOutput>> deviceOutputs;
+    for (const auto& pDevice : std::as_const(m_devices)) {
+        deviceOutputs[pDevice] = m_config.getOutputs().values(pDevice->getDeviceId());
```

Comment:
This avoid the hash lookup before writing:
```suggestion
        deviceOutputs.insert(pDevice, m_config.getOutputs().values(pDevice->getDeviceId()));
```